### ISSUE TITLE
Lyer

### ISSUE THREAD

**Issue by shacklefordjames60-coder** [mfouad/awesome-egypt-opensource#32]
# ... (Prior imports/fixtures)

def test_error_rate_tuning(model, sample_input, sample_labels):  # New
    """Test tuning selects a rate and reduces loss."""
    val_loader = torch.utils.data.DataLoader(
        [(sample_input[i:i+1], sample_labels[i:i+1]) for i in range(4)], 1, False
    )
    initial_er = model.error_rate
    best_rate, best_l = model.tune_error_rate(val_loader, steps=3)
    assert 0.001 <= best_rate <= 0.1, "Rate out of bounds"
    assert best_rate != initial_er or best_l < 1.0, "Should improve or match"
    # Re-forward uses tuned rate
    logits_tuned = model(sample_input)
    assert logits_tuned.shape == (4,)

def test_forward_uses_tuned_rate(model, sample_input):  # New
    """Forward defaults to self.error_rate."""
    h = torch.randn(4, model.hidden_dim)
    q_default = model.quantum_forward_cos(h)
    q_explicit = model.quantum_forward_cos(h, model.error_rate)
    assert torch.allclose(q_default, q_explicit, atol=1e-6)

# Integrate into existing tests
# e.g., in test_qec_mitigation, use tuned rate post-tune
